integration-test-groups/foundation/scheduler/src/main/java/org/apache/camel/quarkus/component/scheduler/it/ShoppingList.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.scheduler.it;
+
+import java.util.*;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Named;
+
+@ApplicationScoped
+@Named("shoppingList")
+public class ShoppingList {
+    List<Integer> lists = new ArrayList<>();
+
+    public void addItems() {
+        int item = new Random().nextInt(100);
+        lists.add(item);
+    }
+
+    public void removeItems() {
+        if (lists != null && lists.size() > 0) {
+            lists.remove(lists.size() - 1);
+        }
+    }
+
+    public List<Integer> getItems() {
+        return lists;
+    }
```

Comment:
There is a potential race between `getItems()` and `addItems()` - you may get concurrent modification exception when accessing the list via `getItems()`. You may want to switch from ArrayList to java.util.concurrent.CopyOnWriteArrayList.